tests/unit/test_charm.py

```diff
@@ -387,3 +387,99 @@ def test_datasource(
 
     harness.update_relation_data(mysql_relation_id, "mysql-k8s", relation_data)
     assert harness.charm._datasource() == expected_output
+
+
+@pytest.mark.parametrize(
+    "files, expected",
+    [
+        (
+            [
+                "/layers/paketo-buildpacks_bellsoft-liberica/jre/bin/java",
+                "/workspace/org/springframework/boot/loader/JarLauncher.class",
+                "/workspace/BOOT-INF/lib/mysql-connector-j-10.1.1.jar",
+            ],
+            True,
+        ),
+        (
+            [
+                "/layers/paketo-buildpacks_bellsoft-liberica/jre/bin/java",
+                "/workspace/org/springframework/boot/loader/JarLauncher.class",
+            ],
+            False,
+        ),
+    ],
+)
+def test_buildpack_app_mysql_capability(
+    harness: Harness, patch: SpringBootPatch, files: list[str], expected: bool
+) -> None:
+    """
+    arrange: provide a simulated OCI image mimicking a Spring Boot application image created by
+        buildpack.
+    act: start the charm.
+    assert: Spring Boot charm should finish the reconciliation process without an error.
+    """
+    app = OCIImageMock.builder()
+    for file in files:
+        app.add_file(file, b"")
+    patch.start({"spring-boot-app": app.build()})
+    harness.begin_with_initial_hooks()
+    assert harness.charm._detect_mysql_capability() == expected
+
+
+@pytest.mark.parametrize(
+    "files, expected",
+    [
+        (
+            [
```

Comment:
Similar to above, could this be a tuple?

Reply:
Fixed